File: data/code_pairs/fortran/1496.f90
program prog
    implicit none
    integer :: i
    integer, dimension(2) :: nums
    integer :: numx
    integer, dimension(4) :: numse

    common /scr/ nums, numx
    equivalence (nums, numse(1)), (numx, numse(3))

    do i = 1, 4
        numse(i) = 10 + i
    end do

    print *, nums(2), nums(1), numse(4), numx
    print *, numse
end program prog

File: data/code_pairs/cpp/1496.cpp
#include <iostream>

struct SharedMemory {
    union {
        int nums[2];
        int numse[4];
    };
    int& numx = nums[1]; // Reference to the second element of nums to mimic FORTRAN COMMON/EQUIVALENCE
};

SharedMemory scr;

int main() {
    // Mimicking the do-loop from Fortran code
    for (int i = 0; i < 4; ++i) {
        scr.numse[i] = 10 + i + 1; // +1 to adjust for 0-based indexing in C++
    }

    // Mimicking the WRITE statements from Fortran code
    std::cout << scr.nums[1] << " " << scr.nums[0] << " " << scr.numse[3] << " " << scr.numx << std::endl;
    for (int i = 0; i < 4; ++i) {
        std::cout << scr.numse[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}